### ISSUE TITLE
images with no objects

### ISSUE THREAD

**Issue by marireeves** [stomperusa/boundingbox#2]
Hi Again. Thank you for your message yesterday and for creating this package. I did get the images to flow with your suggestions from this morning (thank you!), but at the end of the run, I got this error message: 

Error in if (response == 1) { : missing value where TRUE/FALSE needed

I had a few photos where I couldn't find any targets (so no bounding boxes in those, so just moved to the next). Do you think that caused the error?
Please let me know.
This was the code I ran: 

```{r, Define Directories}
TrainPicDir<- 'C:/Users/marireeves/Documents/Lehua/ImagesTrain/'
TrainPicsLabeled<- 'C:/Users/marireeves/Documents/Lehua/TrainLabeled/'
TestPicDir<-  'C:/Users/marireeves/Documents/Lehua/ImagesTest/'
PredPicDir<-  'C:/Users/marireeves/Documents/Lehua/ImagesPred/'
```

```{r, Make a List of Training Image Files}
trainpics<-list.files(TrainPicDir, full.names = F, pattern = "*.png")

```

```{r, Test Bounding Box Function}
#https://cran.r-project.org/web/packages/boundingbox/vignettes/boundingbox-vignette.html


classy<-boxer(
names = trainpics[1:10],
file_path_input =TrainPicDir,
color = "red",
resize_x = NA,
resize_y = NA,
classifier = "Bird",
batch = 5,
outbox = TRUE,
file_path_output = tempdir(),
show_classifier = TRUE
)

```

**Comment by marireeves** [comment]
Update: I've been running in RStudio as a markdown document, so I didn't see the message in the console at first, telling me to hit 1 if I wanted to continue, a different key if not. So I think that was that error message, but when it does run the code chunk and goes to generate the output, I just get the spinning wheel of waiting. Do you think I should run it in R instead of within a markdown ?

**Comment by stomperusa** [comment]
Hi,

So the problem is my instruction says “Enter 1 if you want to truncate the stream or any other key to continue”. What it is looking for is a numerical value. I should have said “…any other number key to continue.” R takes 1 to be TRUE and any other number to be FALSE.  That is why you got a message “missing value where TRUE/FALSE is needed” assuming you entered something other than a number. 

I will put through a fix to clarify this prompt.  In the meantime, enter a number other than 1 to continue the stream. 

I also noted you were saving the output to tempdir(), which I did in my example. I had to do this for the example as CRAN requires that repeatable examples that accompany package documentation do not impact the user's environment.  Unless you have a reason for wanting to work with tempdir(), you may want to have the output saved to a more accessible location, for example a sub-directory of the directory where you have your original files. 

Best Regards,

DS

**Comment by stomperusa** [comment]
I haven’t run it in Rmarkdown before.  I think the issue may be that you are saving the output to tempdir() when Rmarkdown  will run in its own environment. I would suggest changing the location of the output file.  If that doesn’t work, I would run it as a script and trouble shoot from there. 

Regards,

DS

**Comment by marireeves** [comment]
I tried that, actually. And it's been doing that spinning wheel thing for an hour. 

I'm going to try it in a normal RStudio IDE, and let you know how it works. 

Where did you run it from? R Studio or plain R?

```{r, Define Directories}
TrainPicDir<- 'C:/Users/marireeves/Documents/Lehua/ImagesTrain/'
TrainPicsLabeled<- 'C:/Users/marireeves/Documents/Lehua/TrainLabeled/'
TestPicDir<-  'C:/Users/marireeves/Documents/Lehua/ImagesTest/'
PredPicDir<-  'C:/Users/marireeves/Documents/Lehua/ImagesPred/'
```

```{r, Make a List of Training Image Files}
trainpics<-list.files(TrainPicDir, full.names = F, pattern = "*.png")

```

```{r, Test Bounding Box Function}
#https://cran.r-project.org/web/packages/boundingbox/vignettes/boundingbox-vignette.html


classy<-boxer(
names = trainpics[1:10],
file_path_input =TrainPicDir,
color = "red",
resize_x = NA,
resize_y = NA,
classifier = "Bird",
batch = 1,
outbox = TRUE,
file_path_output = TrainPicsLabeled,
show_classifier = TRUE
)
N

**Comment by marireeves** [comment]
It worked in R-studio running as a normal script and I am so excited about it for our use case, that I thought you might be interested and would share since you wrote the package and helped me get it running. 

So, I am trying to use it to label bird nests on the island of Lehua (northwest of Hawaii) in 2 cm resolution drone imagery. ALOT of birds here and alot of images. We're interested in training an object detector to identify nests and a classifier to count the different types of birds that nest here. Super labor intensive work.

I got the photo tiles from the biologist, and tried to use them directly, quickly realizing they were way too dense of information to process by some of the easier means out there. I have an AI for Earth grant from Microsoft, and first tried in their custom vision api, but there were too many birds per photo, so I couldn't upload a photo tile to predict after identifying lots of birds per photo and training a model, because the file size was too large. I'm sure it's surmountable there, but it was also really hard to rigorously examine each photo at that size for the small targets we are trying to identify.

So I parsed each tile into 64 images using the image_slicer package in Python. https://pypi.org/project/image-slicer/

Then I started looking for other packages out there to develop the training set (I am better at R than Python) so found your package. 

I do think it's going to work for our use case, where I can probably send a biologist the R script and have them label a test set of images! Thank you so much for developing and contributing the package.

Mari

**Comment by stomperusa** [comment]
I am glad you got everything to work properly. Thank you for sharing the details of your project. If you are working with multiple classifiers in the same stream of images, you may want to try boxer2().  I hope boundingbox helps make some of that labor intensive work easier to address.

DS
